src/main.c

```diff
@@ -288,31 +291,67 @@ static uint32_t read_encoder_right(void)
 
 
 /**
+ * @brief Setup for adc: Four injected channels on ADC1.
+ */
+static void setup_adc(void)
+{
+	int i;
+	uint8_t channel_array[4];
+
+	/* Make sure the ADC doesn't run during config. */
+	adc_power_off(ADC1);
+
+	adc_enable_scan_mode(ADC1);
+	adc_set_single_conversion_mode(ADC1);
+	/* We want to start the injected conversion in SW*/
+	adc_enable_external_trigger_injected(ADC1, ADC_CR2_JEXTSEL_JSWSTART);
+
+	adc_set_right_aligned(ADC1);
+	adc_set_sample_time_on_all_channels(ADC1, ADC_SMPR_SMP_28DOT5CYC);
+
+	/* Select the channels we want to convert.
+	 * 4=Sensor1, 5=Sensor2, 6=Sensor3, 7=Sensor4
+	 */
+	channel_array[0] = 4;
+	channel_array[1] = 5;
+	channel_array[2] = 6;
+	channel_array[3] = 7;
+	adc_set_injected_sequence(ADC1, 4, channel_array);
+	adc_power_on(ADC1);
+
+	/* Wait for ADC starting up. */
+	for (i = 0; i < 800000; i++)    /* Wait a bit. */
```

Comment:
Maybe we can remove the "wait a bit" comment (repeated).

Reply:
Maybe we could say "wait about 3 microseconds". We could add a note in the docstring pointing to the `adc_power_on` documentation (where they suggest to use this wait).

By the way, how much are we waiting with that 800000?

Reply:
I took that value from libopencm3 official examples for ADC (stm32/f1/lisa-m-2/adc_injec etc).
I guess that, if the clock is at 72 MHz, the nop takes 1 cycle, and it is called 800000. 
(800000/(72*10^6) = 0.01 s)

Reply:
Comments updated ;)